store/mongo/datastore_mongo.go

```diff
@@ -343,7 +343,7 @@ func (db *DataStoreMongo) GetDeviceAuthsByIdentityData(ctx context.Context, idat
 
 	c := s.DB(ctx_store.DbFromContext(ctx, DbName)).C(DbDevicesColl)
 
-	filter := bson.M{"id_data": idata}
+	filter := bson.M{"deviceidentity": idata}
```

Comment:
it would be better to construct filter using DeviceAuth struct, imo
filter := &model.DeviceAuth{
        DeviceIdentity: idata,
}
that way we can detect change in the field name - it will not compile when field name will be changed

Reply:
Fixed